pkg/container/kubernetes/client.go

```diff
@@ -1019,6 +1029,19 @@ func configureContainer(
 		WithStdin(attachStdio).
 		WithTTY(false).
 		WithEnv(envVars...)
+
+	// Add container security context if not already present
+	if container.SecurityContext == nil {
```

Comment:
We may want to do other checks too. Currently if users have specified a `securityContext` but say only added `allowPrivilegeEscalation: false`, this check technically wouldn't be true and we'd skip over it. We might want to add more checks so that if `securityContext != nil`, we check for specific fields and add the sensible defaults if they aren't provided.

For example:
User specifies:
```yaml
securityContext:
   allowPrivilegeEscalation: false
```

We should read that but apply
```yaml
securityContext:
   allowPrivilegeEscalation: false
   runAsUser: 1000
   runAsGroup: 1000
    ...
    ....
```

If that makes sense